### ISSUE TITLE
Test templates have wrong character encoding

### ISSUE THREAD

**Issue by bjornen77** [dotnet/test-templates#358]
If a test project is created using the .NET 8 SDK:

dotnet new xunit

The file UnitTest1.cs is not encoded using utf-8. This have several drawbacks for example:

1/ When trying to add comments that have special characters, for example:
`// This is a nice test method 😀 `

an error is displayed in Visual Studion when saving the file.
![unicode](https://github.com/dotnet/test-templates/assets/58444816/d59f0203-5c31-47e7-8fa2-c85773666a24)

2/ But more important, if a unit test is written in that file, the behavior could be different between systems.

The follwing test work if run on Windows:

```
public class UnitTest1
{
    [Fact]
    public void Test1()
    {
        byte[] expected = [0xC3, 0xA5];
        string swedishCharacter = "å";  // utf-8 encoding 0xC3 0xA5

        Assert.Equal(expected, UTF8Encoding.UTF8.GetBytes(swedishCharacter));
    }
}
```
But when running the same test on Linux(WSL), or on build server(linux) it fails:

```
Assert.Equal() Failure
           ↓ (pos 0)
Expected: [195, 165]
Actual:   [239, 191, 189]
           ↑ (pos 0)
```

If the file is re-saved using utf-8, the test in Linux starts to work. 

I think that all templates files shall be utf-8 by default to avoid these kind of problems?

**Comment by cvpoienaru** [comment]
Hello, I had a look and test-templates are not aware of encoding. I am able to repro the issue using VS (judging by the print screen, you do as well), but opening the same file in VS Code or Notepad++ poses no such issue as the file is encoded as Unicode there. This makes me think it's a VS setup issue.

By any chance, do you have any sort of `.editorconfig` file that may set the encoding ? I would open an issue on the VS side using the `Help` -> `Send Feedback` -> `Report a Problem` channel since there's nothing we can do on test-templates side.

**Comment by bjornen77** [comment]
I think that the test templates do have the wrong encoding. If you use the SDK to create a new project then check encoding in linux using file command:

1/ dotnet new xunit =>
file UnitTest1.cs
UnitTest1.cs: ASCII text, with CRLF line terminators

2/ dotnet new console =>
file Program.cs
Program.cs: Unicode text, UTF-8 (with BOM) text, with CRLF line terminators

The reason why it works if you open the file in VS Code is that VS Code automatically saves the file as UTF-8 without giving the visual studio warning.

**Comment by bjornen77** [comment]
"By any chance, do you have any sort of .editorconfig file that may set the encoding ? "

I do not have any .editorconfig. But if i add one with the following everything works as expected.
charset = utf-8

The problem is that if you create a unit test like the example above, then no warning is given in visual studio that the file will be saved with wrong enconding, and the unit test will fail. If you then open the file in VS Code and save it, the unit test will pass.

**Comment by bjornen77** [comment]
Maybe this is a Visual Studio issue after all. Could it be that due to the fact that the default template does not contain an UTF8-BOM(but the Program.cs do for some reason) then Visual Studio defaults to some other encoding than utf-8? 

I think that normally UTF8 should not have BOM, but i think that this is why Visual Studio treats the files differently. But other programs defaults to UTF8 in both cases..

**Comment by Evangelink** [comment]
Reopening as it seems that dotnet/sdk is using UTF8-BOM https://github.com/dotnet/sdk/blob/main/template_feed/Microsoft.DotNet.Common.ItemTemplates/content/Enum-CSharp/Enum1.cs

![image](https://github.com/dotnet/test-templates/assets/11340282/04290c2d-d3dc-45ca-bfd5-94c8f7166133)

@cvpoienaru Could you please take care of the update?

**Comment by Evangelink** [comment]
Fixed by #361.
